internal/dag/dag.go

```diff
@@ -928,7 +928,10 @@ func (s *ServiceCluster) Rebalance() {
 // Secret represents a K8s Secret for TLS usage as a DAG Vertex. A Secret is
 // a leaf in the DAG.
 type Secret struct {
-	Object *v1.Secret
+	Object         *v1.Secret
+	ValidTLSSecret *SecretValidationStatus
```

Comment:
I guess this has the wrapper around the error since we need the distinction between not yet set, no error, and error